tests/backup/backup_test.go

```diff
@@ -1951,6 +1953,108 @@ var _ = Describe("{BackupRestoreOverPeriodSimultaneous}", func() {
 	})
 }*/
 
+func generateEncryptionKey() string {
+	var lower = []byte("abcdefghijklmnopqrstuvwxyz")
+	var upper = []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	var number = []byte("0123456789")
+	var special = []byte("~=+%^*/()[]{}/!@#$?|")
+	allChar := append(lower, upper...)
+	allChar = append(allChar, number...)
+	allChar = append(allChar, special...)
+
+	b := make([]byte, 12)
+	// select 1 upper, 1 lower, 1 number and 1 special
+	b[0] = lower[rand.Intn(len(lower))]
+	b[1] = upper[rand.Intn(len(upper))]
+	b[2] = number[rand.Intn(len(number))]
+	b[3] = special[rand.Intn(len(special))]
+	for i := 4; i < 12; i++ {
+		// randomly select 1 character from given charset
+		b[i] = allChar[rand.Intn(len(allChar))]
+	}
+
+	//shuffle character
+	rand.Shuffle(len(b), func(i, j int) {
+		b[i], b[j] = b[j], b[i]
+	})
+
+	return string(b)
+}
+
+// This test cases creates a backup location with encryption
+var _ = Describe("{BackupLocationWithEncryptionKey}", func() {
+	var contexts []*scheduler.Context
+	var appContexts []*scheduler.Context
+	var bkpNamespaces []string
+	var backupLocationName string
+	var CloudCredUID string
+	var clusterUid string
+	var CredName string
+	var restoreName string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	JustBeforeEach(func() {
+		StartTorpedoTest("Backup: BackupLocationWithEncryptionKey", "Creating BackupLoaction with Encryption Keys", nil, 0)
+	})
+	It("Creating cloud account and backup location", func() {
+		log.InfoD("Creating cloud account and backup location")
+		providers := getProviders()
+		bucketName := getBucketName()
+		CredName = fmt.Sprintf("%s-%s", "cred", providers[0])
+		backupLocationName = fmt.Sprintf("%s-%s", "location", providers[0])
+		CloudCredUID = uuid.New()
+		BackupLocationUID = uuid.New()
+		encryptionKey := generateEncryptionKey()
+		CreateCloudCredential(providers[0], CredName, CloudCredUID, orgID)
+		CreateBackupLocation(providers[0], backupLocationName, BackupLocationUID, CredName, CloudCredUID, bucketName[0], orgID, encryptionKey)
```

Comment:
Here `bucketName` parameter should be `bucketName := fmt.Sprintf("%s-%s", provider, bucket_name[0])` so You would want to add this line above.